monoio-http/src/common/macros.rs

```diff
@@ -0,0 +1,107 @@
+#[macro_export]
+macro_rules! impl_cookie_extractor {
+    ($struct:ident, $cookie_str:literal) => {
+        impl<P> $struct<P> {
+            /// Builds a cookie jar from the $struct's headers.
+            pub fn build_cookie_jar(&mut self) -> Result<CookieJar, ExtractError> {
+                let headers = self.inner.headers();
+                let mut jar = CookieJar::new();
+
+                if let Some(cookie_header) = headers.get($cookie_str) {
```

Comment:
Using cookiejar seems not needed? I think the purpose of cookiejar is to continuously track cookie status from a session perspective. But for us, we just need to read it.

Reply:
I agree, CookieJar is not required right now. It just looked like a clean way to do it. I thought it might be useful in the future too, when we want to get deltas, track cookies. etc for request and response.